projects/update_ping_ready_nightly_validation.kp/knowledge.md

```diff
@@ -0,0 +1,347 @@
+3. Check how many clients from the list at (1) are in the list at (2).
+
+**Step 1** - Get the list of client ids updating to build '20170809xxxxxx'
+
+
+```python
+TARGET_BUILDID_MIN = '20170809000000'
+TARGET_BUILDID_MAX = '20170809999999'
+
+update_candidates =\
+    deduped_subset.filter(lambda p: TARGET_BUILDID_MIN <= p.get("payload/targetBuildId") <= TARGET_BUILDID_MAX)
+update_candidates_clientIds = dedupe(update_candidates, "clientId").map(lambda p: p.get("clientId"))
+candidates_count = update_candidates_clientIds.count()
+```
+**Step 2** - Get the `main-ping` from that Nightly build and extract the list of client ids.
+
+
+```python
+updated_main_pings = Dataset.from_source("telemetry") \
+    .where(docType="main") \
+    .where(appUpdateChannel="nightly") \
+    .where(submissionDate=lambda x: "20170809" <= x < "20170816") \
+    .where(appBuildId=lambda x: "20170809" <= x < "20170810") \
+    .records(sc, sample=1)
+```
+    fetching 9871.08945MB in 1135 files...
+
+
+We just need the client ids and a few other fields to dedupe.
+
+
+```python
+subset_main = get_pings_properties(updated_main_pings, ["id",
+                                                        "clientId",
+                                                        "meta/Timestamp",
+                                                        "application/buildId",
+                                                        "application/channel",
+                                                        "application/version"])
+```
+Start by deduping by document id. After that, only get a single ping per client and extract the list of client ids.
+
+
+```python
+deduped_main = dedupe(subset_main, "id")
+updated_clientIds = dedupe(deduped_main, "clientId").map(lambda p: p.get("clientId"))
+updated_count = updated_clientIds.count()
+```
+**Step 3** - Count how many clients that were meant to update actually updated in the folliwing 7 days.
+
+
+```python
+matching_clientIds = update_candidates_clientIds.intersection(updated_clientIds)
+matching_count = matching_clientIds.count()
+```
+
+```python
+print("{:.3f}% of the clients that sent the update ping updated to the newer Nightly build within a week."\
+      .format(pct(matching_count, candidates_count)))
+print("{:.3f}% of the clients that were seen on the newer Nightly build sent an update ping."\
+      .format(pct(candidates_count, updated_count)))
+```
+    93.419% of the clients that sent the update ping updated to the newer Nightly build within a week.
+    79.678% of the clients that were seen on the newer Nightly build sent an update ping.
+
+
+Roughly 80% of the clients that were seen in the new Nightly build also sent the update ping. Since we're just waiting 7 days for the data to come in, and we know that it might take up to 5 days (see [here](https://sql.telemetry.mozilla.org/queries/6576) and [here](https://sql.telemetry.mozilla.org/queries/6575)), this seems to be in the ballpark.
```

Comment:
Doesn't take 5 days for Nightly main ping delay. Nightly 57 95%ile is seeing a 9.4 hour delay according to that first link.